### DISCUSSION TITLE
Is it possible to align synchronized charts? (AreaChart and BarChart)

### DISCUSSION THREAD

**Discussion by FranciscoBraaz** [recharts/recharts#4567] [General]
Hi,

I'm using Recharts to build two charts that need to be synchronized. Currently they look like this:
![correct-img1](https://github.com/recharts/recharts/assets/50181942/56c408c4-1956-485f-83d7-bbc648e20e5f)

I need the x-axes to be aligned. Like this:
![correct-img2](https://github.com/recharts/recharts/assets/50181942/861ce9dc-f9d1-48dd-b3dc-a24f2366deb6)

Searching here on the forum I found this issue:
1-) https://github.com/recharts/recharts/issues/1616

I tried to reproduce the stackoverflow answer, but I didn't get the expected result (the first and/or last bar are cut). Here is the example:
https://codesandbox.io/p/sandbox/synchronized-line-charts-forked-v5j979

**Comment by ckifer** [comment]
There is nothing internal to recharts that synchronizes axes or aligns the synchronized charts. The only things that get synced are
1. Tooltip
2. Cursor
3. Active elements
4. Data shown via Brush


Things that could be done to make it seem like this is happening:

1. add padding on your `XAxis` - [example](https://codesandbox.io/p/sandbox/synchronized-line-charts-forked-v5j979). This will never be perfect alignment, only make it seem closer.
2. change the scale of one of your charts - [example](https://codesandbox.io/p/sandbox/synchronized-line-charts-forked-v5j979). Change the BarChart `XAxis` to have `scale="point"` - this cuts off the bars, but looks like what you have in your screenshot

**Comment by FranciscoBraaz** [comment]
Got it, thanks for the answer
